### PR TITLE
notify and document file array

### LINKED ISSUE #39: Data binding files for other element attribute
[https://jsbin.com/pukibiviqa/edit?html,js,output](https://jsbin.com/pukibiviqa/edit?html,js,output)

vaadin-upload's `files` property initial value does not work with data binding.
In the example when I add one file and then remove it, data binding works as expected.

### LINKED ISSUE #40: Data binding files to dom-repeat
[https://jsbin.com/dunusixuti/edit?html,js,output](https://jsbin.com/dunusixuti/edit?html,js,output)

Data binding `files` for `dom-repeat` does not work.
`Uncaught TypeError: Cannot read property 'concat' of undefined`

### PR DESCRIPTION
**Pull request by manolo**
Fixes #39 
Fixes #40

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-upload/48)

### CODE REVIEW

**Comment by tomivirkki** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-upload.html, line 265 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt2v8ne82hle8joI8t:-KCt2v8ne82hle8joI8u:2017494300) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L265)):*
"Bind to this property if you want to show some custom info in your UI" sounds vague. I'd leave this out.

---

*[vaadin-upload.html, line 268 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt1r8PMKQ-wRWu6WJQ:-KCt1r8QUMJofOzi5Y3w:1446409035) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L268)):*
This is actually a bit controversial. vaadin-upload modifies File objects by adding custom properties to them. File is expected to have a standard set of properties so this is a bit odd.

A cleaner approach would be to wrap the actual File object like:

{
file: File,
speed: Number
....
}

Not really related to this PR though

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by tomivirkki** [comment]
*[vaadin-upload.html, line 57 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt3Q9xatQUNNrFxlav:-KCt3Q9xatQUNNrFxlaw:-1967550531) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L57)):*
iron-list ?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-upload.html, line 57 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt3Q9xatQUNNrFxlav:-KCtAeKI9EKJVVT5xBUG:485502058) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L57)):*
Sorry, already removed in second amend, was here by mistake, I used iron list as well as repeat template to full test data binding to the files property

---

*[vaadin-upload.html, line 265 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt2v8ne82hle8joI8t:-KCtAvG_DrRKGhrkP0NM:-1986003843) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L265)):*
Yes, I'm thinking about how to better document the property. Also I guess we need to think in the gwt api, so probably for all these objects we need some approach like in the grid.

---

*[vaadin-upload.html, line 268 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt1r8PMKQ-wRWu6WJQ:-KCtBFVkIodDJbz7ypKd:1611990514) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L268)):*
Well, it's quite common to add data to JS objects. But yes could be done in a different PR

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by tomivirkki** [comment]
*[vaadin-upload.html, line 271 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCtC03V4V0rSbM62WjG:-KCtC03V4V0rSbM62WjH:482376919) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L271)):*
elapsed

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[vaadin-upload.html, line 271 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCtC03V4V0rSbM62WjG:-KCtIy0rGW4avOAVXMcv:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L271)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by tomivirkki** [comment]
I added an event-listener to "files-changed". Dropping 1 file produced 52 events to the listener. Can this be cut down somehow?

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-:-KCuPVn-csxtC66angKu:-1912538253)*

**Comment by platosha** [comment]
@tomivirrki you just need to filter for `event.detail.path === 'files.splices'` in the listener. Otherwise there're all kinds of state modifications for each file on starting upload, on progress, on success, etc...

---

Review status: 0 of 4 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-:-KCufpL3r-OsMyB0B-m2:-1184566282)*

**Comment by manolo** [comment]
I think removing file-add file-remove events should be done in a separated PR.
I'm preparing a PR for that, is ok for you review this without considering that?

---

Review status: 0 of 4 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-:-KCutfHuxLP9ctpSYXpj:-1003074513)*

**Comment by samiheikki** [comment]
Reviewed 1 of 2 files at r6.
Review status: 1 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-upload-file.html, line 250 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCy-PDK_B5rWLjkHYGQ:-KCy-PDK_B5rWLjkHYGR:331699472) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload-file.html#L250)):*
`elemnt`

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by samiheikki** [comment]
Reviewed 1 of 3 files at r4, 1 of 2 files at r6.
Review status: 3 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[demo/advanced.html, line 150 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCy1QIoRrvFSY3VgSY7:-KCy1QIptF_aiGhe-rMM:740392060) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/demo/advanced.html#L150)):*
Should `type` be `accept`?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by samiheikki** [comment]
Reviewed 1 of 2 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-upload.html, line 288 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCy2oB3q2ezwp5UY-jU:-KCy2oB4iO9PjDiM_fUH:913547620) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload.html#L288)):*
trasfering

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by tehapo** [comment]
Reviewed 1 of 3 files at r4, 1 of 2 files at r5, 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-upload-file.html, line 239 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCyg51kUqRCCVAgNpLC:-KCyg51kUqRCCVAgNpLD:1167008166) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload-file.html#L239)):*
Should use a lowercase "L" in the "listened" now that it's not in the beginning of the sentence.

---

*[vaadin-upload.html, line 268 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCygYEP8BnNohxzxDCg:-KCygYEP8BnNohxzxDCh:-494205225) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload.html#L268)):*
I would add quotes around the `files` and `disabled` attribute values here. Not strictly needed, but something we always do in our own code (and also is done in Polymer examples).

---

*[vaadin-upload.html, line 272 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCygudbEon2oMhDl_TQ:-KCygudbEon2oMhDl_TR:-1920695394) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload.html#L272)):*
Unnecessary extra space between "properties" and "to".

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-upload-file.html, line 239 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCyg51kUqRCCVAgNpLC:-KCyoviFpNBYY259tXug:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload-file.html#L239)):*
Done.

---

*[vaadin-upload-file.html, line 250 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCy-PDK_B5rWLjkHYGQ:-KCyoyWYeHz2SY2dpCQB:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload-file.html#L250)):*
Done.

---

*[vaadin-upload.html, line 268 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCygYEP8BnNohxzxDCg:-KCyp76nNA8aVN7IDyxA:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload.html#L268)):*
Done.

---

*[vaadin-upload.html, line 272 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCygudbEon2oMhDl_TQ:-KCypAhxRh6XYtvjXGzM:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload.html#L272)):*
Done.

---

*[vaadin-upload.html, line 288 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCy2oB3q2ezwp5UY-jU:-KCypH33S2tSUO3sgu2m:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/vaadin-upload.html#L288)):*
Done.

---

*[demo/advanced.html, line 150 \[r6\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCy1QIoRrvFSY3VgSY7:-KCypWcvx6C8tdNh_HjS:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/7a2ebaac29fbaf86ef5d4f0a3ffc41ccabc99bca/demo/advanced.html#L150)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by samiheikki** [comment]
Reviewed 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by platosha** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-upload.html, line 187 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCz4UEmp1lg6BPSm_A_:-KCz4UEmp1lg6BPSm_Aa:-676092941) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L187)):*
I think, for native element attributes we should use attribute binding here, namely `accept$=` and `multiple$=` 

---

*[vaadin-upload.html, line 263 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCz3DxNABnc9s0tHCkh:-KCz3DxNABnc9s0tHCki:512345461) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L263)):*
Not sure if there're any reasons to begin with `It is` here. It's the description of the property, no need to point out that, IMHO.

---

*[vaadin-upload.html, line 268 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCz3kcWoMtM0UV12goF:-KCz3kcWoMtM0UV12goG:-553843128) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L268)):*
`disabled` is a native element attribute. It's recommended to use attribute binding (`$=`) in this case. See: https://www.polymer-project.org/1.0/docs/devguide/data-binding.html#native-binding

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by tomivirkki** [comment]
*[vaadin-upload.html, line 265 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt2v8ne82hle8joI8t:-KCzBfBgL5YfiBGXPs8u:-380499588) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L265)):*
What I mean is that "binding" to a property is something you do with frameworks/libraries but it's not a native JavaScript term.

And the "to show some custom info" doesn't really say what it does...maybe "...to affect the file list displayed by the component". But I think the first sentence covers it well enough so I'd prefer leaving out the second sentence.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by tomivirkki** [comment]
*[vaadin-upload.html, line 288 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCzEaupAToqflxE53Pz:-KCzEaupAToqflxE53Q-:-396236256) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L288)):*
transferring

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[vaadin-upload.html, line 265 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCt2v8ne82hle8joI8t:-KCzLWE8U64BpVIsC22U:2102747686) ([raw file](https://github.com/vaadin/vaadin-upload/blob/df8bfbfbf9fa51c02de999821fb80d4a74e80306/vaadin-upload.html#L265)):*
Do I remove the example as well, or it's ok to leave it here?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: 3 of 4 files reviewed at latest revision, 11 unresolved discussions.

---

*[vaadin-upload.html, line 187 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCz4UEmp1lg6BPSm_A_:-KCzMfgXtyfzZrB81Ej_:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L187)):*
Done.

---

*[vaadin-upload.html, line 263 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCz3DxNABnc9s0tHCkh:-KCzMmIP1Ohtm4boKcOu:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L263)):*
Done.

---

*[vaadin-upload.html, line 268 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCz3kcWoMtM0UV12goF:-KCzMn75aigICQzYmcN5:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L268)):*
Done.

---

*[vaadin-upload.html, line 288 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCzEaupAToqflxE53Pz:-KCzMtU9ZsunJw3F6rjO:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/9a8510aa9affdafb169ce69037fc66c1000262c7/vaadin-upload.html#L288)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by platosha** [comment]
Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[demo/advanced.html, line 147 \[r8\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCzYrIM36nMdZzxJk3p:-KCzYrINZbqAV9Qk34b9:1807741028) ([raw file](https://github.com/vaadin/vaadin-upload/blob/cf62bb892bb2c572e3ae253828970b8cf0c61ad4/demo/advanced.html#L147)):*
Please Mind the Title Case

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[demo/advanced.html, line 147 \[r8\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCzYrIM36nMdZzxJk3p:-KCztORzxytH09MNHRWR:-584094813) ([raw file](https://github.com/vaadin/vaadin-upload/blob/cf62bb892bb2c572e3ae253828970b8cf0c61ad4/demo/advanced.html#L147)):*
What words should be Capitalized? I don't know what are important or not, is this ok ?

Pre-Filling The File List In Advance

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by platosha** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[demo/advanced.html, line 147 \[r8\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCzYrIM36nMdZzxJk3p:-KCzvk7qg63TDxQ09Gbq:-453641754) ([raw file](https://github.com/vaadin/vaadin-upload/blob/cf62bb892bb2c572e3ae253828970b8cf0c61ad4/demo/advanced.html#L147)):*
Pre-Filling the File List in Advance

In short, all words are important, except a, an, the, at, by, for, in, of, on, to, up, and, as, but, or, and nor.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: 3 of 4 files reviewed at latest revision, 9 unresolved discussions.

---

*[demo/advanced.html, line 147 \[r8\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KCzYrIM36nMdZzxJk3p:-KCzxERJtKjJC65wuzPK:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/cf62bb892bb2c572e3ae253828970b8cf0c61ad4/demo/advanced.html#L147)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by platosha** [comment]
Reviewed 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[demo/advanced.html, line 153 \[r9\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KD-AE0EdxIfIzpkxl69:-KD-AE0FOEDmw8a9dBxe:2066453152) ([raw file](https://github.com/vaadin/vaadin-upload/blob/ab32233eb5e30b8e310ec8aaba7070abed2e8701/demo/advanced.html#L153)):*
No reasons to capitalise first letter in “novels”.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by samiheikki** [comment]
Reviewed 1 of 1 files at r8, 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[demo/advanced.html, line 153 \[r9\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KD-AE0EdxIfIzpkxl69:-KD2MRyRyNN1GMoSXJpr:-1432396337) ([raw file](https://github.com/vaadin/vaadin-upload/blob/ab32233eb5e30b8e310ec8aaba7070abed2e8701/demo/advanced.html#L153)):*
It's delivered, I wanted to emphasize that the user is selecting Novels, not other kind of books.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by tomivirkki** [comment]
*[test/upload.html, line 34 \[r9\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KD2aPUWJ5WPqgG8qUhS:-KD2aPUWJ5WPqgG8qUhT:162628740) ([raw file](https://github.com/vaadin/vaadin-upload/blob/09881685751dafa8b46bdc4e909aa22c046e422d/test/upload.html#L34)):*
Why is this test removed?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[test/upload.html, line 34 \[r9\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48#-KD2aPUWJ5WPqgG8qUhS:-KD2dzHqZa6aZ3RrkOjy:1631004836) ([raw file](https://github.com/vaadin/vaadin-upload/blob/09881685751dafa8b46bdc4e909aa22c046e422d/test/upload.html#L34)):*
No reason, probably a mistake, or a bad merge/rebase.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*

**Comment by Saulis** [comment]
Reviewed 1 of 3 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/48)*
